processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -572,7 +572,7 @@ private Accessor getAdderForType(Type collectionProperty, String pluralPropertyN
                         continue;
                     }
                     VariableElement arg = executable.getParameters().get( 0 );
-                    if ( arg.asType().equals( typeArg ) ) {
+                    if ( typeUtils.isSameType( arg.asType(), typeArg ) ) {
```

Comment:
`arg.asType()` is a `TypeMirror` and not our `Type`. `arg` is also not our object but  `javax.lang.model.element.VariableElement`